### PR TITLE
fix(onboard): restore portable authority on resume

### LINKED ISSUE #9035: Persist portable runtime authority across onboarding resume
Parent: #9006

## Investigation Summary

- Portable preparation currently writes only process-local runtime selectors.
- Plain or automatic resume does not restore the portable profile before profile-dependent decisions.
- The onboarding state machine resumes after completed steps, so it can skip the preparation that selected the rootless Podman endpoint.
- Checkpoint schema v3 stores no profile or runtime authority.

## Description

Persist one secret-free portable onboarding intent and rootless Podman authority in the onboarding checkpoint. On every resume, reconstruct and requalify the current user's canonical runtime before any resumed runtime decision or resource mutation. Reject profile, identity, path, ownership, endpoint, or checkpoint drift without Docker or ambient-variable fallback.

This issue owns durable onboarding resume authority only. Broad native-Podman lifecycle conversion remains outside scope.

## Reproduction Steps

1. Start `nemoclaw onboard --experimental-profile portable --fresh`.
2. Force a controlled failure after a completed onboarding boundary.
3. Start a new process without the prepared `DOCKER_HOST`, `CONTAINERS_CONF`, and `NETAVARK_FW` values.
4. Run `nemoclaw onboard --resume`.
5. Observe that current `main` does not reconstruct the portable profile/runtime. Explicit portable plus resume is rejected.

## Acceptance Criteria

- [ ] Plain resume reconstructs portable intent from a strict checkpoint; explicit portable plus resume must match.
- [ ] Resume rederives config and requalifies the same current-user rootless Podman endpoint after process restart or reboot.
- [ ] Hostile ambient runtime selectors cannot redirect onboarding.
- [ ] Drift, tampering, unsafe ownership/path state, stale endpoint, Docker substitution, and active legacy checkpoints without recorded authority fail before writes.
- [ ] Session identity, completed steps, resources, and unrelated leases remain stable across retry and failure.
- [ ] Deterministic boundary, concurrency, and race tests pass.
- [ ] Clean-Linux rootless-Podman resume, status, policy, and chat evidence passes.

## Environment

- Linux
- Rootless Podman 5.7.0
- OpenShell 0.0.101
- NemoClaw `main`

## Checklist

- [x] I confirmed this bug is reproducible from the source flow and lifecycle validation.
- [x] I searched existing issues and this is not a duplicate.

### PR DESCRIPTION
**Pull request by senthilr-nv**
## Summary

Portable onboarding resume now reconstructs its rootless Podman authority from a versioned checkpoint instead of inheriting process-local runtime selectors. Resume restores the recorded portable profile before admission, requalifies the canonical current-user socket and configuration under the onboarding lock, and fails closed on legacy, tampered, unsafe, or drifting authority.

Live status: **PARTIAL PASS / BLOCKED DOWNSTREAM BY #9068**. The live lane proved #9035 authority reconstruction and socket requalification from an active v4 checkpoint with a pre-existing Ready sandbox; it did not reach the policy boundary, 8/8 completion, chat, completed-resume, or full live acceptance.

## Related Issue

Fixes #9035

Parent: #9006

Downstream blocker: #9068

## Changes

- Store checkpoint schema 4 portable intent and secret-free current-user Podman authority, with exact cross-field validation and intentional active schema 1–3 resume refusal.
- Pre-read only the resume profile classification, re-read the exact session under the lifecycle lock, permit one bounded race retry, then prepare and requalify portable authority before resumed runtime consumers.
- Recompute canonical current-user home, configuration, runtime, socket, and managed configuration settings while rejecting profile, UID, path, type, owner, symlink, endpoint, runtime-kind, or authority drift.
- Clear ambient Docker, Podman, XDG, inference, and policy selectors during scoped execution and restore their exact prior presence and values on every return, throw, and handled exit.
- Preserve upstream #8992 gateway-database recovery semantics: portable incompatible-state recovery prints the explicit portable `--fresh` command, while default recovery prints `--resume`, in both guidance branches.
- Preserve upstream #9075's timeout-argument parsing source, tests, and command reference byte-for-byte while integrating the current `main` history.
- Document portable resume, fail-closed recovery, canonical roots, and the active legacy-checkpoint compatibility break.

The live resume accepted a reboot-like socket inode rotation while preserving the exact endpoint, owner, mode, immutable authority digest, session identity, Ready sandbox/container identity, and registry identity. After #9035-owned qualification, the run entered #9068's Docker GPU-patch/recreation path and unexpectedly attempted to stop the forward for unrelated sandbox `my-assistant`; that is downstream behavior, not expected or normalized #9035 behavior. The preserved lane was not patched, migrated, retried around the boundary, or claimed as full acceptance.

## Type of Change

- [ ] Code change (feature, bug fix, or refactor)
- [x] Code change with doc updates
- [ ] Doc only (prose changes, no code sample modifications)
- [ ] Doc only (includes code sample changes)

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [x] Docs updated for user-facing behavior changes
- [ ] Docs not applicable — justification:
- [x] Sensitive paths changed (security, policy, credentials, preflight, onboarding, inference, runner, sandbox, or messaging)
- [x] Sensitive-path review completed or maintainer-approved waiver recorded — reviewer/approval link/justification: maintainer task `019f5e28-0f70-7313-92ef-40a3233f796e` approved the schema, lock ordering, environment transaction, security review, scoped live waiver, final diff, and publication.
- [ ] Non-success, skipped, or missing CI check accepted by maintainer — check name, approval link, and follow-up issue:

## Documentation Writer Review

- [x] Documentation writer subagent reviewed the completed changes
- Result: `docs-updated`
- Evidence: Independently reviewed signed automatic merge head `52d026093f15dcb75b82c65f61f5ba01c8e63e4e` from #9035 parent `338493906f3ac2abc96e2049b2e124c1314897da` and upstream parent `267abe79856c96fcf74477bd3ce803a942f286fd`. Exact-head validation passed: base-aware Vitest 326 files/4,209 tests; focused #9035 resume/security 8 files/122 tests; upstream Jetson/OpenShell/MCP boundary 3 files/34 tests; CLI typecheck; repository checks (1,698 files/5,205 edges/0 cycles); conditionals; docs CLI parity 84/84, starter, variants, and routes; onboard growth +59/-62; and diff checks. The exported exact HEAD source-shape scan passed with 0 new cases, 0 invalid exceptions, and 119 approved exceptions. The composed command reference preserves portable resume/fresh recovery and upstream destroy-identity documentation.
- Agent: Codex Desktop

## DGX Station Hardware Evidence

- [ ] Tested on DGX Station
- Tested commit: Not applicable; `scripts/prepare-dgx-station-host.sh` is unchanged.
- Station profile/scenario: Not applicable.
- Result: Not applicable.
- Supporting evidence: Not applicable.

## Verification

- [x] PR description includes a `Signed-off-by:` line and every commit appears as `Verified` in GitHub
- [x] Normal `pre-commit`, `commit-msg`, and `pre-push` hooks passed, except the maintainer-approved root source-shape hook skip caused by unrelated untracked nested repositories. The exact committed tree passed the same source-shape check with 0 new cases, 0 invalid exceptions, and 119 approved existing exceptions.
- [x] Targeted behavior tests pass for the current change set, or tests are marked not applicable above — focused #9035 resume/security: 8 files/122 tests; upstream Jetson/OpenShell/MCP boundary: 3 files/34 tests.
- [x] Applicable broad gate passed — base-aware Vitest against `upstream/main`: 326 files/4,209 tests; `npm run typecheck:cli`; `npm run checks:repository`.
- [x] Quality Gates section completed with required justifications or waivers
- [x] No secrets, API keys, or credentials committed
- [ ] `npm run docs` builds without warnings (doc changes only)
- [x] Doc pages follow the [style guide](https://github.com/NVIDIA/NemoClaw/blob/main/docs/CONTRIBUTING.md) (doc changes only)
- [ ] New doc pages include SPDX header and frontmatter (new pages only)

The final merge commit used:

```sh
PREK_SKIP='tmp/llama-cpp-priority-profiles,tmp/sagecove-llamacpp-poc-full,.:source-shape-test-budget' git commit -S -s -m 'merge(main): integrate upstream changes'
```

Merge and auto-merge are not authorized. This scoped publication waiver does not claim merge readiness while #9068 blocks protected downstream acceptance.

---
Signed-off-by: Senthil Ravichandran <senthilr@nvidia.com>

## Summary by CodeRabbit

* **New Features**
  * Improved onboarding resume support with checkpoint validation, profile compatibility checks, and portable runtime restoration.
  * Added safer portable runtime handling, including filesystem, socket, ownership, and permission validation.
  * Added clearer recovery guidance with separate resume and fresh-onboarding commands.
* **Bug Fixes**
  * Prevented unsafe environment settings from affecting portable onboarding.
  * Added bounded retry handling for resume conflicts and restored environment state after failures.
  * Older checkpoint schemas now require fresh onboarding instead of automatic migration.
* **Documentation**
  * Expanded guidance for portable profiles, checkpoint compatibility, runtime authority validation, and recovery.

### CODE REVIEW

File: src/lib/onboard/experimental/portable-host-preparation.ts [hunk 1]
```diff
@@ -231,7 +422,34 @@ export function preparePortableExperimentalHost(
     podman(["info", "--format", "{{.Host.RemoteSocket.Path}}"], podmanEnv),
   );
   const socketPath = dockerHost.slice("unix://".length);
+  if (expectedAuthority && socketPath !== expectedAuthority.socketPath) {
```
Comment (apurvvkumaria) [inline src/lib/onboard/experimental/portable-host-preparation.ts:425]: **Non-blocking under this review threshold:** this live endpoint comparison runs after `writePortableRuntimeConfig()` has written three files and after all three `systemctl --user` mutations. I reproduced a valid v4 authority with a different missing socket descendant: the files were created and all three systemctl calls ran before this mismatch threw. Live Podman qualification also occurs later, so runtime substitution can fail only after those mutations. This conflicts with #9035’s fail-before-writes acceptance criterion, although I did not identify critical security impact. Please follow up by separating read-only resume admission from remediation and add negative tests asserting zero config/systemctl effects on endpoint mismatch and qualification failure.

**Review by apurvvkumaria — APPROVED** [review]
Approved: I found no critical-severity blocker after reviewing checkpoint integrity, locking, environment scoping, Podman authority, recovery paths, documentation, and focused tests.

I left one explicitly non-blocking note about host mutation preceding live requalification. Also non-blocking: `resolveOnboardResumeIntent()` parses the saved session before honoring `--fresh`, so malformed JSON requires manual deletion instead of the documented fresh recovery path.
